WorldModel/WorldModel/Core/CoreOutput.aslx

```diff
@@ -874,14 +874,16 @@
 
   <function name="Log" parameters="text">
     <![CDATA[
-    //request (Log, text)
+    request (Log, text)
     // Replacing double quotes with 2 single quotes
     text = Replace(text, "\"", "''")
     // Changing syntax from single quotes to escaped double quotes to allow single quotes in log entries
     if (not GetBoolean(game, "nohtmllog")){
       JS.eval("if(typeof(addLogEntry)===\"function\"){ addLogEntry(\""+text+"\"); };")
     }
-    JS.eval("if(!webPlayer && typeof(WriteToLog)===\"function\"){var s = \""+text+"\";WriteToLog(s);}")
+    if (GetBoolean(game, "writelogtofile") and StartsWith(GetFileURL("foo"),"quest")){
+      JS.eval("if(typeof(WriteToLog)===\"function\"){var s = \"" + text + "\";WriteToLog(s);}")

```

Comment:
I know this is just updating what is here already, but I'm not such a fan of the `eval`s here - could this call a JS function instead?

Reply:
Should I add an `IsDesktop` function to make that IF statement look better, too? (Also, do you know of a better way to check that than `GetFileURL`?)

Reply:
I got rid of all this by adding a dummy WriteToLog in player.js.